R/library.R

```diff
@@ -272,6 +279,20 @@ download_package.github <- function(package, force) {
   remote_download.github_remote(remote, dest, quiet = !isTRUE(getOption('lockbox.verbose')))
 }
 
+## download a raw tarball from a URL
+download_package.tarball <- function(package, force) {
```

Comment:
Why do we need both a downloading method and an installing method, when the installing method doesn't call out to the download method? (likely just that I don't know the layout of the lockbox package)

Reply:
@abelcastilloavant This is a good point. However, note in the PR that the method dispatch for `get_remote_dependencies` for the tarball is the same as for CRAN, thus calling out to `download_package`.

Reply:
I don't understand why we download the packages (in order to get dependencies and all that) and then for the installation part we completely ignore the downloaded package and instead install from CRAN, github or URL directly, thereby redownloading the package. Seems like a little refactoring would help. I'm lost in the codebase myself, frankly